src/Workspaces/CSharp/Portable/CodeGeneration/CSharpSyntaxGenerator.cs

```diff
@@ -2770,45 +2823,31 @@ private AccessorDeclarationSyntax AsAccessor(SyntaxNode node, SyntaxKind parentK
 
         private AccessorDeclarationSyntax GetAccessor(SyntaxNode declaration, SyntaxKind kind)
         {
-            switch (declaration.Kind())
-            {
-                case SyntaxKind.PropertyDeclaration:
-                    return ((PropertyDeclarationSyntax)declaration).AccessorList.Accessors.FirstOrDefault(a => a.IsKind(kind));
-                case SyntaxKind.IndexerDeclaration:
-                    return ((IndexerDeclarationSyntax)declaration).AccessorList.Accessors.FirstOrDefault(a => a.IsKind(kind));
-                default:
-                    return null;
-            }
+            var accessorList = GetAccessorList(declaration);
+            return accessorList?.Accessors.FirstOrDefault(a => a.IsKind(kind));
         }
 
         private SyntaxNode WithAccessor(SyntaxNode declaration, SyntaxKind kind, AccessorDeclarationSyntax accessor)
         {
-            switch (declaration.Kind())
-            {
-                case SyntaxKind.PropertyDeclaration:
-                    return WithAccessor(declaration, ((PropertyDeclarationSyntax)declaration).AccessorList, kind, accessor);
-                case SyntaxKind.IndexerDeclaration:
-                    return WithAccessor(declaration, ((IndexerDeclarationSyntax)declaration).AccessorList, kind, accessor);
-                default:
-                    return declaration;
-            }
+            return WithAccessor(declaration, GetAccessorList(declaration), kind, accessor);
         }
 
         private SyntaxNode WithAccessor(SyntaxNode declaration, AccessorListSyntax accessorList, SyntaxKind kind, AccessorDeclarationSyntax accessor)
         {
-            var acc = accessorList.Accessors.FirstOrDefault(a => a.IsKind(kind));
-            if (acc != null)
-            {
-                return this.ReplaceNode(declaration, acc, accessor);
-            }
-            else if (accessor != null)
-            {
-                return this.ReplaceNode(declaration, accessorList, accessorList.AddAccessors(accessor));
-            }
-            else
+            if (accessorList != null)
```

Comment:
So if you call WithAccessor on an expression bodied property, we just drop it on the floor?  Is there any way to convert from expression bodied to accessored?